### PR TITLE
Add resource: aws_networkmanager_dx_gateway_attachment

### LINKED ISSUE #40310: Cloud WAN integration with Direct Connect
> AWS Cloud WAN now supports native integration with AWS Direct Connect, simplifying connectivity between your on-premises networks and the AWS cloud.

* [Announcement](https://aws.amazon.com/about-aws/whats-new/2024/11/aws-cloud-wan-on-premises-connectivity-direct-connect/)
* [Blog post](https://aws.amazon.com/blogs/networking-and-content-delivery/simplify-global-hybrid-connectivity-with-aws-cloud-wan-and-aws-direct-connect-integration/)

### Affected Resource(s) and/or Data Source(s)

* `aws_networkmanager_dx_gateway_attachment`

### PR DESCRIPTION
**Pull request by ddericco**
### Description

- Adds a new resource `aws_networkmanager_dx_gateway_attachment` to enable support for Direct Connect gateway attachments to a Cloud WAN core network.
- Updates the `aws_networkmanager_attachment_accepter` resource to support Direct Connect gateway attachments.

**NOTES:**
- The default account quota for global networks (and Cloud WAN core networks by extension) is 5 - if running the acceptance tests for this resource, consider setting a lower `ACCTEST_PARALLELISM` or requesting a [quota increase](https://us-east-1.console.aws.amazon.com/servicequotas/home/services/networkmanager/quotas/L-2418390E).
- Unlike VPC or other core network attachments, the Direct Connect gateway attachment can only be updated when the state is `AVAILABLE` - all other states, including Pending Attachment Acceptance, return an InvalidInput error.

### Relations

Closes #40310

### References

- API reference: https://docs.aws.amazon.com/networkmanager/latest/APIReference/API_CreateDirectConnectGatewayAttachment.html
- Go SDKv2: https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/networkmanager
- What's New: https://aws.amazon.com/about-aws/whats-new/2024/11/aws-cloud-wan-on-premises-connectivity-direct-connect/

### Output from Acceptance Testing

```console
% make testacc TESTS='TestAccNetworkManagerDXGatewayAttachment_*' PKG=networkmanager

make testacc TESTS='TestAccNetworkManagerDXGatewayAttachment_*' PKG=networkmanager ACCTEST_PARALLELISM=3
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/networkmanager/... -v -count 1 -parallel 3 -run='TestAccNetworkManagerDXGatewayAttachment_*'  -timeout 360m
2024/12/12 15:58:37 Initializing Terraform AWS Provider...
=== RUN   TestAccNetworkManagerDXGatewayAttachment_basic
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_basic
=== RUN   TestAccNetworkManagerDXGatewayAttachment_disappears
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_disappears
=== RUN   TestAccNetworkManagerDXGatewayAttachment_update
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_update
=== CONT  TestAccNetworkManagerDXGatewayAttachment_basic
=== RUN   TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_required
=== CONT  TestAccNetworkManagerDXGatewayAttachment_update
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_required
=== RUN   TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_not_required
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_not_required
=== CONT  TestAccNetworkManagerDXGatewayAttachment_disappears
=== CONT  TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_required
=== RUN   TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_required
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_required
=== RUN   TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_not_required
=== PAUSE TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_not_required
=== CONT  TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_not_required
=== CONT  TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_required
=== CONT  TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_not_required
--- PASS: TestAccNetworkManagerDXGatewayAttachment_basic (0.00s)
    --- PASS: TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_required (565.63s)
    --- PASS: TestAccNetworkManagerDXGatewayAttachment_basic/acceptance_not_required (1011.83s)
--- PASS: TestAccNetworkManagerDXGatewayAttachment_disappears (0.00s)
    --- PASS: TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_required (526.46s)
    --- PASS: TestAccNetworkManagerDXGatewayAttachment_disappears/acceptance_not_required (912.59s)
--- PASS: TestAccNetworkManagerDXGatewayAttachment_update (2746.25s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager     2754.021s
```

### CODE REVIEW

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% ACCTEST_TIMEOUT=720m make testacc TESTARGS='-run=TestAccNetworkManagerDirectConnectGatewayAttachment_\|TestAccNetworkManagerConnectAttachment_basic\|TestAccNetworkManagerSiteToSiteVPNAttachment_basic\|TestAccNetworkManagerTransitGatewayRouteTableAttachment_basic\|TestAccNetworkManagerVPCAttachment_basic' PKG=networkmanager ACCTEST_PARALLELISM=2
make: Verifying source code with gofmt...
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go1.23.3 test ./internal/service/networkmanager/... -v -count 1 -parallel 2  -run=TestAccNetworkManagerDirectConnectGatewayAttachment_\|TestAccNetworkManagerConnectAttachment_basic\|TestAccNetworkManagerSiteToSiteVPNAttachment_basic\|TestAccNetworkManagerTransitGatewayRouteTableAttachment_basic\|TestAccNetworkManagerVPCAttachment_basic -timeout 720m
2024/12/16 12:13:14 Initializing Terraform AWS Provider...
=== RUN   TestAccNetworkManagerConnectAttachment_basic
=== PAUSE TestAccNetworkManagerConnectAttachment_basic
=== RUN   TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
=== PAUSE TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_basic
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_basic
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_disappears
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_disappears
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_update
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_update
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_tags
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_tags
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_accepted
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_accepted
=== RUN   TestAccNetworkManagerSiteToSiteVPNAttachment_basic
=== PAUSE TestAccNetworkManagerSiteToSiteVPNAttachment_basic
=== RUN   TestAccNetworkManagerTransitGatewayRouteTableAttachment_basic
=== PAUSE TestAccNetworkManagerTransitGatewayRouteTableAttachment_basic
=== RUN   TestAccNetworkManagerVPCAttachment_basic
=== PAUSE TestAccNetworkManagerVPCAttachment_basic
=== CONT  TestAccNetworkManagerConnectAttachment_basic
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_tags
--- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_tags (1005.82s)
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_disappears
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_required
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_required
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_not_required
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_not_required
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_update
--- PASS: TestAccNetworkManagerConnectAttachment_basic (1109.07s)
=== CONT  TestAccNetworkManagerTransitGatewayRouteTableAttachment_basic
--- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_update (2238.14s)
=== CONT  TestAccNetworkManagerVPCAttachment_basic
=== RUN   TestAccNetworkManagerVPCAttachment_basic/acceptance_required
=== PAUSE TestAccNetworkManagerVPCAttachment_basic/acceptance_required
=== RUN   TestAccNetworkManagerVPCAttachment_basic/acceptance_not_required
=== PAUSE TestAccNetworkManagerVPCAttachment_basic/acceptance_not_required
=== CONT  TestAccNetworkManagerSiteToSiteVPNAttachment_basic
--- PASS: TestAccNetworkManagerTransitGatewayRouteTableAttachment_basic (1531.48s)
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_accepted
--- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_accepted (887.94s)
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_basic
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_required
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_required
=== RUN   TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_not_required
=== PAUSE TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_not_required
=== CONT  TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
--- PASS: TestAccNetworkManagerSiteToSiteVPNAttachment_basic (1079.18s)
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_required
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_not_required
=== NAME  TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
    connect_attachment_test.go:66: Error running post-test destroy, there may be dangling resources: exit status 1
        
        Error: deleting Network Manager VPC Attachment (attachment-0262efda0afdc290e): operation error NetworkManager: DeleteAttachment, https response error StatusCode: 400, RequestID: 8d6a0109-6960-439a-b1b8-5bc6ae9f131d, ValidationException: VPC attachment (attachment-0262efda0afdc290e) cannot be deleted due to existing Connect attachment (attachment-0e9d2f9034e1685b5).
        
--- FAIL: TestAccNetworkManagerConnectAttachment_basic_NoDependsOn (833.94s)
=== CONT  TestAccNetworkManagerVPCAttachment_basic/acceptance_required
=== CONT  TestAccNetworkManagerVPCAttachment_basic/acceptance_not_required
--- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_disappears (0.00s)
    --- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_required (564.14s)
    --- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_disappears/acceptance_not_required (1004.15s)
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_required
=== CONT  TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_not_required
--- PASS: TestAccNetworkManagerVPCAttachment_basic (0.00s)
    --- PASS: TestAccNetworkManagerVPCAttachment_basic/acceptance_required (589.31s)
    --- PASS: TestAccNetworkManagerVPCAttachment_basic/acceptance_not_required (898.66s)
--- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_basic (0.00s)
    --- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_required (540.92s)
    --- PASS: TestAccNetworkManagerDirectConnectGatewayAttachment_basic/acceptance_not_required (921.80s)
FAIL
FAIL	github.com/hashicorp/terraform-provider-aws/internal/service/networkmanager	6678.836s
FAIL
make: *** [testacc] Error 1
```

Failure is unrelated to this change and is occurring in CI (e.g. https://hashicorp.teamcity.com/buildConfiguration/TerraformProviders_Aws_Main_ServiceTest_networkmanager/244942):

```
=== RUN   TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
=== PAUSE TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
=== CONT  TestAccNetworkManagerConnectAttachment_basic_NoDependsOn
    connect_attachment_test.go:66: Error running post-test destroy, there may be dangling resources: exit status 1
        Error: deleting Network Manager VPC Attachment (attachment-065991be773c7877e): operation error NetworkManager: DeleteAttachment, https response error StatusCode: 400, RequestID: 5888622e-4be8-496b-82c1-5075a0b76887, ValidationException: VPC attachment (attachment-065991be773c7877e) cannot be deleted due to existing Connect attachment (attachment-0d7b6f1cef89825bd).
--- FAIL: TestAccNetworkManagerConnectAttachment_basic_NoDependsOn (895.11s)
```

File: website/docs/r/networkmanager_dx_gateway_attachment.html.markdown [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+[Configuration options](https://developer.hashicorp.com/terraform/language/resources/syntax#operation-timeouts):
+
+* `create` - (Default `30m`)
+* `update` - (Default `30m`)
+* `delete` - (Default `30m`)
+
+## Import
+
+In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import Network Manager DX Gateway Attachment using the `example_id_arg`. For example:
```
Comment (jar-b) [inline website/docs/r/networkmanager_dx_gateway_attachment.html.markdown:58]:
```suggestion
In Terraform v1.5.0 and later, use an [`import` block](https://developer.hashicorp.com/terraform/language/import) to import Network Manager DX Gateway Attachment using the `id`. For example:
```

File: website/docs/r/networkmanager_dx_gateway_attachment.html.markdown [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+```terraform
+import {
+  to = aws_networkmanager_dx_gateway_attachment.example
+  id = "attachment-1a2b3c4d5e6f7g"
+}
+```
+
+Using `terraform import`, import Network Manager DX Gateway Attachment using the `example_id_arg`. For example:
```
Comment (jar-b) [inline website/docs/r/networkmanager_dx_gateway_attachment.html.markdown:67]:
```suggestion
Using `terraform import`, import Network Manager DX Gateway Attachment using the `id`. For example:
```

File: website/docs/r/networkmanager_dx_gateway_attachment.html.markdown (hunk 1 above)
Comment (jar-b) [reply website/docs/r/networkmanager_dx_gateway_attachment.html.markdown:58]: Also need to document `id` in the attribute reference above.

**Review by jar-b — COMMENTED** [review]
A couple doc nits, otherwise LGTM

**Review by jar-b — APPROVED** [review]
LGTM 🚀

**Comment by ewbankkit** [comment]
@ddericco Thanks for the contribution :tada: :clap:.

**Pull request merged**